### DISCUSSION TITLE
How to install/update ImageMagick to a newer version on Shared Hosting

### DISCUSSION THREAD

**Discussion by gtaillefer** [ImageMagick/ImageMagick#5644] [Help]
Hello, in a [recent post](https://github.com/ImageMagick/ImageMagick/discussions/5632), I found out that in order to solve a problem that I'm having, I need to update ImageMagick to a newer version (7.1.0 49 or later). However, as shown in the post, I am on a Bluehost/cpanel shared hosting, on CentOS 6. The version of ImageMagick that I'm using is 6.9.10-68. I discovered a [while back](https://stackoverflow.com/questions/73683467/downloading-imagemagick-for-mediawiki-error-failed-dependencies) that trying to rpm the libraries or the rpm file of ImageMagick resulted in errors saying that I didn't have many of the dependencies needed. Since then I had figured out the solution to the problem that I was having with ImageMagick (at the time, which was to find the older version of a program that I was using which fit the glibc dependencies required), so it never needed to be resolved. However now I'm stuck with the issue again and still no idea how to solve it. Since I'm on shared hosting, I can't use yum or apt-get to install packages that would solve this problem. 
So therefore I would like to ask, is there any way to update ImageMagick to 7.1.0 47 on CentOS 6 shared hosting? Or get around the restrictions to solve this problem of updating ImageMagick? Thanks

**Comment by fmw42** [comment]
If your ISP allows you to install another version other than the one they provide, you might try the portable version.  See the  "magick" version at https://imagemagick.org/script/download.php

**Reply by gtaillefer** [reply]
Ill have to look into that to try it out

**Reply by gtaillefer** [reply]
Unfortunately, since Bluehost wants me to hate them even more than I do, they don't even support FUSE. Trying to install it would be impossible and take time to try and figure out another way to use sudo, yum, and apt-get commands to try and install it. I'm sure that there is a way, but spending even more days and hours trying to figure that out is wasteful, and I'll try to look into it another time when I have more time. Otherwise it seems that the AppImage alternative would work better. For now though I guess I'll just have to stick and get used to this tiff problem. Big thanks for all of your guys' help though

**Reply by fmw42** [reply]
Perhaps one of the IM developers can offer some suggestions.

**Reply by gtaillefer** [reply]
I'm assuming that you are? If so, and even if not that would be great

**Reply by fmw42** [reply]
No, I am not.

**Comment by snibgo** [comment]
> ... I need to update ImageMagick to a newer version (7.1.0 47 or later).

No. I said *don't* use 7.1.0-47 if you are reading TIFF files. If you have that version, I suggest you upgrade to 7.1.0-49 or later.

**Reply by gtaillefer** [reply]
My apologies, I meant to say 49 and above, my mistake.
